src/app/(auth)/reset-password/page.tsx

```diff
@@ -22,29 +24,37 @@ export default function ResetPassword() {
   }, []);
 
   const resetPassword = async () => {
+    if (newPassword !== newPassword2) {
+      setErrorMessage('Passwords do not match.');
+      return;
+    }
+    setErrorMessage('');
     const { error } = await supabase.auth.updateUser({ password: newPassword });
     if (error) {
-      throw new Error(
-        `An error occurred trying to reset password: ${error.message}`,
-      );
-    }
-    const { error: signOutError } = await supabase.auth.signOut();
-    if (signOutError) {
-      throw new Error(
-        `An error occurred trying to sign out: ${signOutError.message}`,
-      );
+      setErrorMessage(error.message);
+    } else {
+      const { error: signOutError } = await supabase.auth.signOut();
+      if (signOutError) {
+        throw new Error(
+          `An error occurred trying to sign out: ${signOutError.message}`,
+        );
+      }
+      push('/confirm-reset-password');
     }
-    push('/confirm-reset-password');
   };
 
   return (
     canReset && (
       <>
-        <H1>Set New Password</H1>
-        <QuestionsDiv>
+        <SpacerDiv $gap={0.625}>
+          <H1>Set New Password</H1>
+          {errorMessage !== '' && <P $color={COLORS.redMid}>{errorMessage}</P>}
+        </SpacerDiv>
+        <SpacerDiv>
           <TextInput
             label="New Password"
             placeholder="Password"
+            errorText={newPassword.length < 6 ? 'Invalid Password' : ''}
```

Comment:
i think we should only set this when the user click on the button, since it could be weird to see an error while the user is still typing out their password

tho also a design question

Reply:
personally, i think this is chill bc ultimately it will be replaced with password complexity stuff -- which will show up as the user types in the password. but yes def design question, cc @davidqing6432

Reply:
i'll leave it as is for now!